### PR TITLE
 JWWEB-875 - Edit contact

### PR DESCRIPTION
**Pull request by flexxnn**

### CODE REVIEW

File: src/pages/ContactsItemEdit/components/ContactEditForm/ContactEditForm.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  description: string, // note
+  name: string,
+|}
+
+export type Props = {|
+  +initialValues: FormValues,
+  +i18n: I18nType,
+  +checkContactExists: (address: OwnerAddress) => boolean,
+  +editContact: (contact: FormValues) => any,
+  +onDeleteClick: (address: OwnerAddress) => any,
```
Comment (IViol) [inline src/pages/ContactsItemEdit/components/ContactEditForm/ContactEditForm.js]: I suggest to name handlers in the same way, e.g. editContact/deleteContact or onEdit/onDelete, etc

File: src/store/router/routes.js [hunk 2]
```diff
@@ -60,15 +60,15 @@ export const routes: Array<{|
   name: 'ContactsItem',
   hasMenu: true,
 }, {
-  path: '/contacts/add?:address&:name&:note',
+  path: '/contacts/add?:address&:name',
```
Comment (IViol) [inline src/store/router/routes.js]: why did you remove `note` param?

File: src/pages/ContactsItemEdit/ContactsItemEdit.js [hunk 3]
```diff
@@ -1,26 +1,103 @@
...
+
+function mapStateToProps(
+  state: AppState,
+  { contactId }: OwnProps,
+) {
+  const {
+    address,
+    name = '',
+    description = '',
+  } = selectFavorite(state, contactId || '') || {}
```
Comment (IViol) [inline src/pages/ContactsItemEdit/ContactsItemEdit.js]: Probably, better to `throw` if contactId is empty (actually, it is not possible, because of route's path - `/contacts/edit/:contactId`)

File: src/pages/ContactsItemEdit/components/ContactEditForm/ContactEditForm.js (hunk 1 above)
Comment (stcherenkov) [reply src/pages/ContactsItemEdit/components/ContactEditForm/ContactEditForm.js]: I don't agree. There is strong difference between edit and click on edit button. `onEdit` could mean: on click on edit button, on type in input, on change input, on focus in input and so on.

Comment (stcherenkov) [reply src/pages/ContactsItemEdit/components/ContactEditForm/ContactEditForm.js]: I also suggest we put handler first, description second. So it would all be structured the same way for all the handlers, like: `onClick` (if it is the only element), `onClickEdit`, `onClickDelete`, `onChangeName`, `onFocusIcon` etc.
